### ISSUE TITLE
[Question] How to start daemon via ShellCommand?

### ISSUE THREAD

**Issue by nosuchip** [buildbot/buildbot#4905]
I have worker that pulls ExpressJS-base API servers, starts it in container and perform number of tests against started server. I'd like to do that. But cannot find how to start daemon via ShellCommand? I'm ok if it would be started in "uncontrolled" state - started and forgotten (everything working in container so background processes will be killed on test finish).

I tried to start it as `nohup node server.js &`. But it doesn't works.

Any idea how to do that?

**Comment by j-rodman** [comment]
Indeed.  backgrounding doesn't work.  Launching by the linux  daemonize command doesn't work. 

Looking at the worker source, It seems that implementation is a posix process group, which should be possible to escape by a classic posix daemonization (double fork, with the precise order of exits)  I do launch a commercial  program that does this work, and it survives  the ShellCommand, but the daemonize command line utility seems insufficiently complete in its implementation.

Since it's not worth writing my own implemenation of daemonization into every thing I'll need to run, I'm left with likely workarounds:   launch indirectly.  That is, run a command that asks an already existing process to launch what your test needs.  Examples would be to ask the os to launch a service out of init / systemd /etc , or to ask dockerd to bring up a container, both of which launch out of a  process outside anything buildbot is trying to control, dodging its remit.   If you need to knock up something more custom, a long running httpd that runs things (and shuts them down) on request is probably the easiest path.

**Comment by p12tic** [comment]
@j-rodman Just a quick question since I haven't re-read into how exactly process groups work: would turning process groups off help your use case?
